ai/scripts/search_test.py

```
@@ -0,0 +1,376 @@
+                    (CASE WHEN full_reference ILIKE :kw2 THEN 2.0 ELSE 0.0 END) +
+                    (CASE WHEN content ILIKE :kw0 THEN 1.0 ELSE 0.0 END) +
+                    (CASE WHEN content ILIKE :kw1 THEN 1.0 ELSE 0.0 END) +
+                    (CASE WHEN content ILIKE :kw2 THEN 1.0 ELSE 0.0 END)
+                ) / GREATEST(:keyword_count * 3.0, 1.0) +
+                0.7 * (1 - (embedding <=> CAST(:query_vector AS vector)))
+            ) AS score
+        FROM {table}
+        WHERE embedding IS NOT NULL
+        ORDER BY score DESC
+        LIMIT :top_k
+    """)
```

Ruff (0.14.0)

Possible SQL injection vector through string-based query construction

(S608)